k9mail/src/main/java/com/fsck/k9/message/quote/TextQuoteCreator.java

```
@@ -0,0 +1,83 @@
+        } else if (quoteStyle == QuoteStyle.HEADER) {
+            StringBuilder quotedText = new StringBuilder(body.length() + QUOTE_BUFFER_LENGTH);
+            quotedText.append("\r\n");
+            quotedText.append(resources.getString(R.string.message_compose_quote_header_separator)).append("\r\n");
+            if (originalMessage.getFrom() != null && Address.toString(originalMessage.getFrom()).length() != 0) {
+                quotedText.append(resources.getString(R.string.message_compose_quote_header_from)).append(" ").append(Address.toString(originalMessage.getFrom())).append("\r\n");
+            }
+            if (sentDate.length() != 0) {
+                quotedText.append(resources.getString(R.string.message_compose_quote_header_send_date)).append(" ").append(sentDate).append("\r\n");
+            }
+            if (originalMessage.getRecipients(RecipientType.TO) != null && originalMessage.getRecipients(RecipientType.TO).length != 0) {
+                quotedText.append(resources.getString(R.string.message_compose_quote_header_to)).append(" ").append(Address.toString(originalMessage.getRecipients(RecipientType.TO))).append("\r\n");
+            }
+            if (originalMessage.getRecipients(RecipientType.CC) != null && originalMessage.getRecipients(RecipientType.CC).length != 0) {
+                quotedText.append(resources.getString(R.string.message_compose_quote_header_cc)).append(" ").append(Address.toString(originalMessage.getRecipients(RecipientType.CC))).append("\r\n");
+            }
+            if (originalMessage.getSubject() != null) {
+                quotedText.append(resources.getString(R.string.message_compose_quote_header_subject)).append(" ").append(originalMessage.getSubject()).append("\r\n");
+            }
+            quotedText.append("\r\n");
+
+            quotedText.append(body);
+
+            return quotedText.toString();
+        } else {
+            // Shouldn't ever happen.
```

Then should we throw an error?